src/nbl/video/IDescriptorPool.cpp

```diff
@@ -61,7 +74,7 @@ bool IDescriptorPool::reset()
 {
     const auto& compilerIsRetarded = m_descriptorSetAllocator;
     for (const uint32_t setIndex : compilerIsRetarded)
-        deleteSetStorage(m_allocatedDescriptorSets[setIndex]);
+        deleteSetStorage(setIndex);
 
     m_descriptorSetAllocator.reset();
```

Comment:
if you want to ensure some sanity (as discussed in discord) you can do the following

```cpp
    // something else except for the allocated sets needs to be holding onto the pool, so that
    // we don't have an implicit call to `~IDescriptorPool` on `this` before we get out of the current stackframe  
    assert(getReferenceCount()>m_descriptorSetAllocator.get_allocated_size());
    
    const auto& compilerIsRetarded = m_descriptorSetAllocator;
    for (const uint32_t setIndex : compilerIsRetarded)
        deleteSetStorage(setIndex);

    // the `reset` doesn't deallocate anything, but it rearranges the free address stack in order to
    // not make the future allocated addresses dependent on what happened before a reset
    assert(m_descriptorSetAllocator.get_allocated_size() == 0);
    m_descriptorSetAllocator.reset();
    
    // see the comment at the first assert
    assert(getReferenceCount()>=1);
    return reset_impl();
```

Reply:
Resolved.